### PR TITLE
Move `mc lock` to `mc bucket lock`

### PR DESCRIPTION
**Pull request by poornas**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -277,10 +279,17 @@ github.com/minio/md5-simd v1.1.0 h1:QPfiOqlZH+Cj9teu0t9b1nTBfPbyTl16Of5MeuShdK4=
 github.com/minio/md5-simd v1.1.0/go.mod h1:XpBqgZULrMYD3R+M28PcmP0CkI7PEMzB3U77ZrKZ0Gw=
 github.com/minio/minio v0.0.0-20200714163805-778e9c864f67 h1:xMg87xvLhrw2jeE6xGx/uuFr2iIksfszITGdHhWURnA=
 github.com/minio/minio v0.0.0-20200714163805-778e9c864f67/go.mod h1:0WThPBMICOt5eEGT3XyaUy0cxYmoGhmJEvvRbiVqbYU=
+github.com/minio/minio-go v6.0.14+incompatible h1:fnV+GD28LeqdN6vT2XdGKW8Qe/IfjJDswNVuni6km9o=
```
Comment (kannappanr) [inline go.sum:282]: you need to remove references to `v6` of minio-go

**Comment by klauspost** [comment]
This is could be breaking for scripts. It shouldn't be a big issue, but we should probably mention it somewhere.

**Comment by kannappanr** [comment]
> This is could be breaking for scripts. It shouldn't be a big issue, but we should probably mention it somewhere.

Yes, we will mention it when we make a release

File: cmd/auto-complete.go [hunk 2]
```diff
@@ -256,6 +256,11 @@ var completeCmds = map[string]complete.Predictor{
 	"/tag/remove": s3Completer,
 	"/tag/set":    s3Completer,
 
+	"/bucket/lock/compliance": s3Completer,
+	"/bucket/lock/governance": s3Completer,
+	"/bucket/lock/clear":      s3Completer,
+	"/bucket/lock/info":       s3Completer,
+
```
Comment (harshavardhana) [inline cmd/auto-complete.go:263]: Your forgot to remove 

> 	"/lock":      complete.PredictOr(s3Complete{deepLevel: 2}),

Comment (poornas) [reply cmd/auto-complete.go:262]: done

File: docs/minio-client-complete-guide.md [hunk 3]
```diff
@@ -596,19 +595,13 @@ mc lock myminio/mybucket compliance 30d
...
-```
-mc lock myminio/mybucket
+mc lock myminio/mybucket info
 COMPLIANCE mode is enabled for 30d
 ```
 *Example: Clear object lock configuration for bucket `mybucket`*
 
 ```
-mc lock --clear myminio/mybucket
+mc lock myminio/mybucket clear
```
Comment (harshavardhana) [inline docs/minio-client-complete-guide.md:604]: This doc is incorrect @poornas

File: docs/minio-client-complete-guide.md [hunk 4]
```diff
@@ -596,19 +595,13 @@ mc lock myminio/mybucket compliance 30d
...
 
 ```
-mc lock myminio/mybucket
-COMPLIANCE mode is enabled for 30d
-```
-*Example: Display the object lock configuration for bucket `mybucket`*
-
-```
-mc lock myminio/mybucket
+mc lock myminio/mybucket info
```
Comment (harshavardhana) [inline docs/minio-client-complete-guide.md:598]:
```suggestion
mc bucket lock myminio/mybucket info
```

File: docs/minio-client-complete-guide.md (hunk 3 above)
Comment (harshavardhana) [inline docs/minio-client-complete-guide.md:604]:
```suggestion
mc bucket lock myminio/mybucket clear
```

**Pull request merged**